### PR TITLE
Reduce duplication of font library group headings

### LINKED ISSUE #63502: Reduce duplication of font library group headings
As a follow up to https://github.com/WordPress/gutenberg/pull/63211, let's remove "Fonts" from the "Theme Fonts" and "Custom Fonts" headings. We don't need so much duplication here. 

CleanShot 2024-07-12 at 16 31 07

This would be in line with how the colors are represented: 

CleanShot 2024-07-12 at 16 33 14

### PR DESCRIPTION
**Pull request by akasunil**
## What?

Fixes https://github.com/WordPress/gutenberg/issues/63502

## Screenshots
image

image

### CODE REVIEW

**Review by matiasbenedetto — APPROVED** [review]
Looks good to me

File: packages/edit-site/src/components/global-styles/font-families.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ function FontFamilies() {
 			<VStack spacing={ 4 }>
 				{ themeFonts.length > 0 && (
 					<VStack>
-						<Subtitle level={ 3 }>{ __( 'Theme Fonts' ) }</Subtitle>
+						<Subtitle level={ 3 }>{ __( 'Theme' ) }</Subtitle>
```
Comment (creativecoder) [inline packages/edit-site/src/components/global-styles/font-families.js]: This seems like a place where we should use `_x` and/or a translator comment to provide a context for translation. Translating just the word "Theme" or "Custom" in isolation may be challenging.

```suggestion
						/* translators: Heading for a list of fonts provided by the theme. */
						<Subtitle level={ 3 }>{ _x( 'Theme', 'font source' ) }</Subtitle>
```

File: packages/edit-site/src/components/global-styles/font-families.js [hunk 2]
```diff
@@ -65,9 +65,7 @@ function FontFamilies() {
 				) }
 				{ customFonts.length > 0 && (
 					<VStack>
-						<Subtitle level={ 3 }>
-							{ __( 'Custom fonts' ) }
-						</Subtitle>
+						<Subtitle level={ 3 }>{ __( 'Custom' ) }</Subtitle>
```
Comment (creativecoder) [inline packages/edit-site/src/components/global-styles/font-families.js]:
```suggestion
						/* translators: Heading for a list of fonts installed by the user. */
						<Subtitle level={ 3 }>{ _x( 'Custom', 'font source' ) }</Subtitle>
```

File: packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js [hunk 3]
```diff
@@ -172,7 +172,7 @@ function InstalledFonts() {
 								{ baseThemeFonts.length > 0 && (
 									<VStack>
 										<h2 className="font-library-modal__fonts-title">
-											{ __( 'Theme Fonts' ) }
+											{ __( 'Theme' ) }
```
Comment (creativecoder) [inline packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js]:
```suggestion
											/* translators: Heading for a list of fonts provided by the theme. */
											{ _x( 'Theme', 'font source'  ) }
```

File: packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js [hunk 4]
```diff
@@ -209,7 +209,7 @@ function InstalledFonts() {
 								{ baseCustomFonts.length > 0 && (
 									<VStack>
 										<h2 className="font-library-modal__fonts-title">
-											{ __( 'Custom fonts' ) }
+											{ __( 'Custom' ) }
```
Comment (creativecoder) [inline packages/edit-site/src/components/global-styles/font-library-modal/installed-fonts.js]:
```suggestion
											/* translators: Heading for a list of fonts installed by the user. */
											{ _x( 'Custom', 'font source' ) }
```

File: packages/edit-site/src/components/global-styles/font-families.js [hunk 5]
```diff
@@ -52,7 +52,12 @@ function FontFamilies() {
 			<VStack spacing={ 4 }>
 				{ themeFonts.length > 0 && (
 					<VStack>
-						<Subtitle level={ 3 }>{ __( 'Theme Fonts' ) }</Subtitle>
+						<Subtitle level={ 3 }>
+							{
+								/* translators: Heading for a list of fonts provided by the theme. */
+								_x( 'Theme' )
```
Comment (creativecoder) [inline packages/edit-site/src/components/global-styles/font-families.js]: `_x` requires a string describing the context as the 2nd argument (see https://github.com/WordPress/gutenberg/tree/trunk/packages/i18n#_x).

**Review by creativecoder — APPROVED** [review]
This looks good now, thanks for addressing the issue!

**Pull request merged**